io/io/src/TBufferFile.cxx

```diff
@@ -1937,10 +1938,11 @@ void TBufferFile::WriteArrayDouble32(const Double_t *d, Int_t n, TStreamerElemen
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Write array of n bools into the I/O buffer.
+/// \note Operation is silently skipped in case of underflow or overflow
 
 void TBufferFile::WriteFastArray(const Bool_t *b, Int_t n)
 {
-   if (n <= 0) return;
+   if (n <= 0 || n > std::numeric_limits<Int_t>::max()/static_cast<Int_t>(sizeof(UChar_t))) return;
```

Comment:
```suggestion
   constexpr Int_t overallMaxElements = std::numeric_limits<Int_t>::max()/static_cast<Int_t>(sizeof(UChar_t);
   const Int_t maxElements = overallMaxElements - Length()/static_cast<Int_t>(sizeof(UChar_t);
   if (n <= 0 || n > maxElements)
      return;
```
if we want to be exact.

Reply:
everywhere below, too?

Reply:
Maybe I create a templated function for this?

Reply:
Yes :)